### PR TITLE
Fix merge region doc and refactor

### PR DESCRIPTION
**Pull request by lhy1024**
### What problem does this PR solve? 

Region merge left into right in https://github.com/pingcap/pd/pull/1564.

But [the description](https://github.com/pingcap/pd/tree/master/tools/pd-ctl) in this document is the opposite 

>  "This means a Region can only be merged into left."

### What is changed and how it works?

Fix document and add more comment.

Otherwise, replace `GetEnableOneWayMerge` with `IsOneWayMergeEnabled` and refactor to make it easier to understanding.

### Check List 

Tests 

 - Unit test

### CODE REVIEW

File: tools/pd-ctl/README.md [hunk 1]
```diff
@@ -197,7 +197,7 @@ Usage:
     >> config set split-merge-interval 24h  // Set the interval between `split` and `merge` to one day
     ```
 
-- `enable-one-way-merge` controls the merge scheduler behavior. This means a Region can only be merged into left.
+- `enable-one-way-merge` controls the merge scheduler behavior. This means a Region can only be merged into right.
```
Comment (disksing) [inline tools/pd-ctl/README.md]: I think either `left` or `right` is ambiguous... How about `can only be merged into next Region of it`?

File: server/checker/merge_checker.go [hunk 2]
```diff
@@ -126,18 +131,9 @@ func (m *MergeChecker) Check(region *core.RegionInfo) []*operator.Operator {
...
-		if target == nil || target.GetApproximateSize() > adjacent.GetApproximateSize() {
-			// peer count should equal
-			if len(adjacent.GetPeers()) == m.cluster.GetMaxReplicas() {
-				target = adjacent
-			}
-		}
-	}
-	return target
+		len(adjacent.GetDownPeers()) == 0 && len(adjacent.GetPendingPeers()) == 0 && len(adjacent.GetLearners()) == 0 && // no special peer
+		len(adjacent.GetPeers()) == m.cluster.GetMaxReplicas() // peer count should equal
```
Comment (disksing) [inline server/checker/merge_checker.go]: Nice refactor 👍

File: tools/pd-ctl/README.md (hunk 1 above)
Comment (lhy1024) [reply tools/pd-ctl/README.md]: done

File: server/checker/merge_checker.go [hunk 3]
```diff
@@ -102,9 +102,14 @@ func (m *MergeChecker) Check(region *core.RegionInfo) []*operator.Operator {
...
 	prev, next := m.cluster.GetAdjacentRegions(region)
 
-	target := m.checkTarget(region, next, nil)
-	if !m.cluster.GetEnableOneWayMerge() {
-		target = m.checkTarget(region, prev, target)
+	var target *core.RegionInfo
+	if m.checkTarget(region, next) {
+		target = next
+	}
+	if !m.cluster.IsOneWayMergeEnabled() && m.checkTarget(region, prev) { // allow to merge region right to left
```
Comment (disksing) [inline server/checker/merge_checker.go]: Please update this line too

Comment (disksing) [reply server/checker/merge_checker.go]: The rest LGTM.

Comment (lhy1024) [reply server/checker/merge_checker.go]: done

**Pull request merged**